Add `nvim-tree` plugin integration

jordykoppen opened the pull request:
Hey, I'm back with another addition :)

| Before  | After |
| ------------- | ------------- |
| <img width="944" alt="image" src="https://user-images.githubusercontent.com/24296556/191268818-c1348ba4-6405-4180-86c7-743e206ece34.png"> | <img width="944" alt="image" src="https://user-images.githubusercontent.com/24296556/191268606-a7cf0d52-01b8-464a-91c6-e765c9edaf02.png"> |

It's more of a quick implementation to get rid of the white border in between. So some suggestions on where to apply what colors would be appreciated!

jordykoppen committed:
feat: add nvim-tree integration

antoineco commented:
I'm interested in this as well.

If I may comment on the changes, all this re-indentation probably belongs in a separate PR. Right now it's really difficult to see what has changed and what has just been re-indented.

jordykoppen commented:
Ah, the re-indentation was not intentional. Let me fix it 😄

sam4llis commented:
Hi, thanks for adding to the plugin again!

I agree with @antoineco with respect to the indentation changes. For what its worth, I format using the `sumneko_lua` LSP (with two spaces per indentation level). I think I actually prefer to highlight groups in the first picture (excluding the white border, obviously) as on first glance it looks like file directory highlights already link to the built-in `Directory` highlight group.

If you're interested in adding extra functionality to this integration as well, I believe I've seen some colorschemes have a darker background for the `nvim-tree` window. I'm not sure if this is a built-in functionality inside `nvim-tree`, but it might be worth looking into :) If you don't want to look into this, open up an issue and I'll have a look into it when I have the time!

Once again, thanks for your contributions to the Tundra theme. They're really appreciated, especially when its additions for plugins I don't use!

antoineco commented:
Here are the highlight groups that control the background of nvim-tree windows: https://github.com/sainnhe/gruvbox-material/issues/139#issuecomment-1239121715

jordykoppen committed:
style: fix indentation

jordykoppen committed:
refactor(nvim-tree): remove unnecessary highlight groups

jordykoppen commented:
Thanks for the suggestions guys!

I've fixed the indentation and removed the highlight groups to keep using the `Directory` highlight group. I did however update the background to be darker, and improve border colors:
<img width="912" alt="image" src="https://user-images.githubusercontent.com/24296556/191970087-3358fd7e-0844-45e7-b485-5d84460ff913.png">

antoineco commented on lua/nvim-tundra/stylesheet/arctic.lua:
```diff
@@ -1,5 +1,5 @@
-package.loaded['nvim-tundra.palette.arctic'] = nil
-local cp = require('nvim-tundra.palette.arctic')
+package.loaded["nvim-tundra.palette.arctic"] = nil
+local cp = require("nvim-tundra.palette.arctic")
```
Small suggestion to revert unrelated changes:

```suggestion
package.loaded['nvim-tundra.palette.arctic'] = nil
local cp = require('nvim-tundra.palette.arctic')
```

antoineco commented on lua/nvim-tundra/stylesheet/arctic.lua:
```diff
@@ -41,9 +41,9 @@ stylesheet.syntax = {
-  func  = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
-  type  = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
-  var   = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
+  func = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
+  type = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
+  var = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
```
Small suggestion to revert unrelated changes:

```suggestion
  func  = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
  type  = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
  var   = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
```

antoineco commented on lua/nvim-tundra/stylesheet/arctic.lua:
```diff
@@ -64,6 +64,27 @@ stylesheet.diff = {
+  statusline = { fg = stylesheet.fg.statusline, bg = stylesheet.bg.transparent },
+  symlink = cp.red._400,
+  border = { bg = stylesheet.bg.normal, fg = stylesheet.bg.normal },
+}
```
It seems like only `filetree.filename.normal`, `filetree.bg` and `filetree.border` are actually used in this changeset. Is that intentional?

jordykoppen replied:
@antoineco sharp! I've removed the unused properties, without altering the styling this time 😂

jordykoppen committed:
style: revert wrong style changes

Co-authored-by: Antoine Cotten <hello@acotten.com>

jordykoppen committed:
Update lua/nvim-tundra/stylesheet/arctic.lua

Co-authored-by: Antoine Cotten <hello@acotten.com>

jordykoppen committed:
refactor(nvim-tree): update stylesheet usage

jordykoppen committed:
refactor: remove unused stylesheet properties

sam4llis commented on lua/nvim-tundra/base_configuration.lua:
```diff
@@ -32,6 +32,7 @@ local base_configuration = {
   plugins = {
     lsp = true,
     treesitter = true,
+    nvimtree = true,
```
This should be false by default. I think only 'base' plugins such as native LSP and Treesitter should be enabled by default.

```suggestion
    nvimtree = false,
```

sam4llis commented on lua/nvim-tundra/hl_group/plugins/nvimtree.lua:
```diff
@@ -0,0 +1,10 @@
+
+M.map = function(ss, opts)
+  return {
+    NvimTreeNormal = { fg = ss.filetree.fg, bg = ss.filetree.bg },
+    NvimTreeVertSplit = ss.filetree.border,
```
Change to use base stylesheet groups instead of a new `filetree` group. I've also changed the background colour of the `nvim-tree` window so that there is a less drastic contrast between it and normal windows.

Before: 
<img width="2132" alt="image" src="https://user-images.githubusercontent.com/43378933/192095996-ded839bf-447d-425a-89e8-5e5d147fbfdb.png">

After:
<img width="2132" alt="image" src="https://user-images.githubusercontent.com/43378933/192096056-b5588579-b6f3-4aa7-8ed1-1938f8219582.png">


```suggestion
    NvimTreeNormal = { fg = ss.fg.normal, bg = ss.bg.dimmed },
    NvimTreeVertSplit = { fg = ss.bg.dimmed, bg = ss.bg.dimmed },
```

sam4llis commented on lua/nvim-tundra/stylesheet/arctic.lua:
```diff
@@ -64,6 +64,12 @@ stylesheet.diff = {
   text = cp.indigo._800,
 }
 
+stylesheet.filetree = {
+  bg = stylesheet.bg.floating,
+  fg = stylesheet.fg.normal,
+  border = { bg = stylesheet.bg.normal, fg = stylesheet.bg.normal },
+}
+
```
I think this should be removed from the stylesheet. I want to keep this stylesheet as simple as possible. Check out the suggestion in `nvim-tree.lua` for my proposal!

```suggestion
```

sam4llis reviewed (changes requested):
Looking good! Just a couple of changes that I think should be addressed. Also, it might be worth adding some information about `nvim-tree` support into the `README` of this colorscheme.

jordykoppen committed:
refactor(nvim-tree): disable plugin by default

Co-authored-by: Sam Ellis <43378933+sam4llis@users.noreply.github.com>

jordykoppen committed:
refactor(nvim-tree): remove filetree stylesheet references

Co-authored-by: Sam Ellis <43378933+sam4llis@users.noreply.github.com>

jordykoppen committed:
refactor(nvim-tree): remove filetree from stylesheet

Co-authored-by: Sam Ellis <43378933+sam4llis@users.noreply.github.com>

jordykoppen committed:
docs: add `nvim-tree` plugin details

jordykoppen commented:
@sam4llis I've committed your suggestions and updated the README 👍

sam4llis commented:
Great, I'm happy with this integration now if you are as well? I'll likely add it to the `main` branch for version `0.1` of Tundra, alongside some other niceties :)

jordykoppen commented:
Let's go! 😄
